logback-core/src/main/java/ch/qos/logback/core/joran/conditional/PropertyEvalJavaScriptBuilder.java

```diff
@@ -0,0 +1,94 @@
+package ch.qos.logback.core.joran.conditional;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.ref.WeakReference;
+
+import javax.script.Bindings;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import ch.qos.logback.core.Context;
+import ch.qos.logback.core.spi.PropertyContainer;
+
+public class PropertyEvalJavaScriptBuilder implements Evaluation {
+
+    public static final String CONDITIONAL_JS_ATTR = "conditional-js";
+    private static final String JS_ENGINE_NAME = "JavaScript";
+
+    // variable names provided in JavaScript context
+    static final String PROPERTIES = "props";
+    static final String CONTEXT = "ctx";
+
+    private static WeakReference<ScriptEngine> globalEngine = new WeakReference<ScriptEngine>(null);
+    private Context context;
+    ScriptEngine engine;
+
+    final PropertyContainer localPropContainer;
+
+    PropertyEvalJavaScriptBuilder(PropertyContainer localPropContainer) {
+        this.localPropContainer = localPropContainer;
+        
+        engine = globalEngine.get();
+        if (engine == null) {
+            final ScriptEngineManager factory = new ScriptEngineManager();
+            engine = factory.getEngineByName(JS_ENGINE_NAME);
+            if (engine == null) {
+                throw new IllegalStateException("JavaScript not supported");
+            }
+            final Reader reader = new InputStreamReader(getClass().getResourceAsStream("functions.js"));
```

Comment:
It supplies the actual property accessor functions that are also available via Janino (like `p()` and `isNull()` etc.). Unfortunately, I forget to check that file in. Shame on me! Doing it tonight.